pandas/core/dtypes/cast.py

```diff
@@ -1085,3 +1085,24 @@ def cast_scalar_to_array(shape, value, dtype=None):
     values.fill(fill_value)
 
     return values
+
+
+def _maybe_convert_indexer(indexer, until):
```

Comment:
that's fine. it just belongs in internal utlities related to indexing (and not internal casting type things). yes it is casting but is purely related to indexing.